lib/services/authority.go

```diff
@@ -50,6 +69,105 @@ func NewCertAuthority(
 	})
 }
 
+// ValidateCertAuthority validates the CertAuthority
+func ValidateCertAuthority(ca CertAuthority) (err error) {
+	if err = ca.CheckAndSetDefaults(); err != nil {
+		return trace.Wrap(err)
+	}
+	switch ca.GetType() {
```

Comment:
To be honest, I'm not sure I'm a fan that the resource validation logic is being spread out across multiple functions. Why can't it be a part of CheckAndSetDefaults as before?

Reply:
The checks on `CertAuthority` are dependent on `lib/jwt`, `ssh`, `lib/utils`, and possibly more due to those dependencies' imports. I agree that this might not be the best solution.

I think the solution will be to move all validation logic to services on the server side, but this is a somewhat larger change. Splitting the validation logic for a few problematic types is just a temporary solution to reach MVP.

It'll be much easier to decide on the best solution when all of the teleport dependencies in api are refactored out, and we can initialize the api go mod (not possible now due to cyclic module dependency with teleport).